File: backend/Litres.Domain/Entities/OutboxMessage.cs
namespace Litres.Domain.Entities;

public class OutboxMessage
{
    public string Id { get; set; }
    public string Type { get; set; }
    public string Content { get; set; }
    public DateTime OccuredOn { get; set; }
    public DateTime ProcessedOn { get; set; }
    public string? Error { get; set; }
}

File: backend/Litres.Payment/src/Models/OutboxMessage.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity()
export class OutboxMessage {
  @PrimaryGeneratedColumn()
  id: string;

  @Column('text')
  type: string;

  @Column('text')
  content: string;

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  occuredOn: Date;

  @Column({ type: 'timestamp', nullable: true, default: null})
  processedOn: Date;

  @Column('text')
  error: string;
}